### PR TITLE
Feature/draft fixes

### LINKED ISSUE #242: [dashboard] Filter on Nations Panel (to only show Blockchain total)
## Issue
The total number of Nations in Pangea shown also adds all the DRAFTS Nations the user have. We need to filter that total to be only the Nations currently accepted in the Blockchain.

## Acceptance criteria
- [ ] When the Nations total only shows the nations in Blockchain

### PR DESCRIPTION
**Pull request by seland**
Closes #242, #243.

#### Changes
Fixed isDraft calculation.
Fixed dashboard nations panel to not display drafts.

### CODE REVIEW

File: src/reducers/nations.js [hunk 1]
```diff
@@ -72,4 +72,12 @@ export default function (state = initialState, action) {
 }
 
 export const openedNation = state => resolveNation(state.nations, state.openedNationId);
-export const isDraft = nation => resolveStatus(nation) === 'draft';
+export const isDraft = (nation) => {
```
Comment (florianlenz) [inline src/reducers/nations.js]: Just check if the smart contract's id is `<= -1` and if the `nation.tx === null`. If the tx is null, it mean's that it was not submitted to the blockchain.

File: src/screens/Dashboard/NationsPanel/index.js [hunk 2]
```diff
@@ -22,8 +23,9 @@ const NEWEST_NATION_COUNT = 5;
 export default class NationsPanel extends Component {
   render() {
     const { style } = this.props;
-    const newestNations = _.take(_.sortBy(this.props.nations, nation => -nation.id), NEWEST_NATION_COUNT);
-    const nationsCountStrings = i18n.t('screens.dashboard.nationsPanel.nationsCount', { count: this.props.nations.length });
+    const existedNations = _.filter(this.props.nations, nation => !isDraft(nation));
```
Comment (florianlenz) [inline src/screens/Dashboard/NationsPanel/index.js]: please use an explicit check.

File: src/screens/NationDetailsContainer/index.js [hunk 3]
```diff
@@ -128,7 +128,10 @@ NationDetailsContainer.PropTypes = {
 const mapStateToProps = state => ({
   ...state.nations,
   ...state.wallet,
-  isDraft: isDraft(openedNation(state.nations) || {}),
+  isDraft: (() => {
+    const nation = openedNation(state.nations);
+    return nation ? isDraft(nation) : true;
```
Comment (florianlenz) [inline src/screens/NationDetailsContainer/index.js]: Can we use a explicit check here as well? like `typeof nation === 'object'`

File: src/reducers/nations.js (hunk 1 above)
Comment (seland) [reply src/reducers/nations.js]: I'd like to have all status determination logic in one place, so I move it to utils/nations.js.

File: src/screens/Dashboard/NationsPanel/index.js (hunk 2 above)
Comment (seland) [reply src/screens/Dashboard/NationsPanel/index.js]: Fixed

File: src/screens/NationDetailsContainer/index.js (hunk 3 above)
Comment (seland) [reply src/screens/NationDetailsContainer/index.js]: I use nation !== null to check.
Important that, typeof nation === 'object' is TRUE if nation is null.

Comment (florianlenz) [reply src/screens/NationDetailsContainer/index.js]: Haha nice. Good catch.

**Comment by florianlenz** [comment]
Someone need's to fix coveralls.

**Pull request merged**